✨Compatibility to pyglotaran (0.5.0-dev) staging branch

s-weigand opened the pull request:
This change prevents crashes when using results from `pyglotaran==0.5.0-dev` in cases where the `irf` has multiple centers and dispersion.
It also adds forward compatibility to the name change `center_dispersion_1` -> `irf_center_location`.
The assumption to use `irf_loc = irf_loc.mean()` is, that the center locations of the `irf` only disperse over the spectral dimension and not over `time`.

### Change summary

- Allows usage with multidimensional `irf_loc` (`spectral`, `irf_nr`)

### Checklist

- [x] ✔️ Passing the tests (mandatory for all PR's)

jsnel commented on pyglotaran_extras/plotting/plot_traces.py:
```diff
@@ -13,14 +13,22 @@ def get_shifted_traces(res, center_λ=None):
     else:
         irf_loc = min(times)
 
+    if hasattr(irf_loc, "shape"):
```
Don't take the average of the irf pulses but pass the centers and assume the [0]th element as the default for 'the' irf locaiton.

jsnel reviewed (changes requested):
adjust irf_loc

s-weigand commented:
Addressed requested changes and added some typing and docstring.
The docstrings for the `symlog` options are directly from the maplotlib docs.
https://matplotlib.org/stable/api/scale_api.html#matplotlib.scale.SymmetricalLogScale

jsnel commented on pyglotaran_extras/plotting/plot_traces.py:
```diff
@@ -13,14 +44,22 @@ def get_shifted_traces(res, center_λ=None):
         irf_loc = min(times)
 
+    if hasattr(irf_loc, "shape"):
+        irf_loc = irf_loc[main_irf_nr]
```
Checking for a shape attribute is not sufficient, a xarray dataarray or numpy array may be 0 dimensions (just a scalar) which cannot be indexed resulting in an error `IndexError("too many indices") `

s-weigand replied:
Thanks, nice catch didn't know that the `numpy` scalars also had a `shape` property.

jsnel reviewed (changes requested):
irf_loc may be a scalar

jsnel reviewed (approved):
LGTM
